zerver/views/invite.py

```diff
@@ -135,6 +137,46 @@ def get_user_invites(request: HttpRequest, user_profile: UserProfile) -> HttpRes
     return json_success(request, data={"invites": all_users})
 
 
+@require_member_or_admin
+def get_invite_details(
+    request: HttpRequest, user_profile: UserProfile, invite_id: PathOnly[int]
+) -> HttpResponse:
+    try:
+        prereg_user = PreregistrationUser.objects.get(id=invite_id)
+    except PreregistrationUser.DoesNotExist:
+        raise JsonableError(_("No such invitation"))
+
+    if prereg_user.realm != user_profile.realm:
+        raise JsonableError(_("No such invitation"))
+
+    if prereg_user.referred_by_id != user_profile.id:
+        check_role_based_permissions(prereg_user.invited_as, user_profile, require_admin=True)
```

Comment:
Can you do a preparatory PR factoring this block of logic into a shared function `prereg_user = access_invite_by_id(user_profile, invite_id)`? That would follow our standard patterns for checking permissions etc.